File: MonRoadtrip-back/src/main/java/monRoadtrip/model/Reservation.java
package monRoadtrip.model;

import java.time.LocalDate;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@Entity
public class Reservation {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	
	@Column(name="date_reservation")
	private LocalDate dateReservation;
	
	@Enumerated(EnumType.STRING)
	private Statut statut;
	
	@ManyToOne
	@JoinColumn(name="id_participant_fk")
	private Participant participant;
	
	@OneToMany(mappedBy = "reservation")
	private List<Etape> etapes;
	
	@ManyToOne
	@JoinColumn(name="id_roadtrip_fk")
	private Roadtrip roadTrip;
	
	@ManyToOne
	@JoinColumn(name="id_client_fk")
	private Client client;

	public Reservation() {}
	
	public Reservation(LocalDate dateReservation, Statut statut, Participant participant, Roadtrip roadTrip, Client client) {
		this.dateReservation = dateReservation;
		this.statut = statut;
		this.participant = participant;
		this.roadTrip = roadTrip;
		this.client = client;
	}
	
	public Reservation(Integer id, LocalDate dateReservation, Statut statut, Participant participant, List<Etape> etapes,
			Roadtrip roadTrip, Client client) {
		this.dateReservation = dateReservation;
		this.statut = statut;
		this.participant = participant;
		this.roadTrip = roadTrip;
		this.client = client;
		this.id=id;
		this.etapes=etapes;
	}
	
	public Reservation(LocalDate dateReservation, Statut statut, Participant participant, List<Etape> etapes,
			Roadtrip roadTrip, Client client) {
		this.dateReservation = dateReservation;
		this.statut = statut;
		this.participant = participant;
		this.roadTrip = roadTrip;
		this.client = client;
		this.etapes=etapes;
	}

	public LocalDate getDateReservation() {
		return dateReservation;
	}

	public void setDateReservation(LocalDate dateReservation) {
		this.dateReservation = dateReservation;
	}
	

	public Statut getStatut() {
		return statut;
	}

	public Participant getParticipant() {
		return participant;
	}

	public List<Etape> getEtapes() {
		return etapes;
	}

	public Roadtrip getRoadTrip() {
		return roadTrip;
	}

	public Client getClient() {
		return client;
	}

	public void setStatut(Statut statut) {
		this.statut = statut;
	}

	public void setParticipant(Participant participant) {
		this.participant = participant;
	}

	public void setEtapes(List<Etape> etapes) {
		this.etapes = etapes;
	}

	public void setRoadTrip(Roadtrip roadTrip) {
		this.roadTrip = roadTrip;
	}

	public void setClient(Client client) {
		this.client = client;
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	@Override
	public String toString() {
		return "Reservation [id=" + id + ", dateReservation=" + dateReservation + ", statut=" + statut + ", participant=" + participant
				+ ", etapes=" + etapes + ", client=" + client + "]";
	}


	
	

}

File: monRoadTripAngular/src/app/model/reservation.ts
import { Roadtrip } from './roadtrip';
import { Participant } from './participant';
import { Client } from './client';
import { Statut } from './statut';
import { Etape } from './etape';
export class Reservation {
  public constructor(
    private _id?: number | undefined,
    private _dateReservation?: Date | undefined,
    private _statut?: Statut | undefined,
    //private _etape?: Etape | undefined,
    private _client?: Client | undefined,
    private _participant?: Participant | undefined,
    private _roadtrip?: Roadtrip | undefined
  ) {}

  public get id(): number | undefined {
    return this._id;
  }

  public set id(value: number | undefined) {
    this._id = value;
  }

  public get dateReservation(): Date | undefined {
    return this._dateReservation;
  }

  public set dateReservation(value: Date | undefined) {
    this._dateReservation = value;
  }

  public get statut(): Statut | undefined {
    return this._statut;
  }

  public set statut(value: Statut | undefined) {
    this._statut = value;
  }
  /*
  public get etape(): Etape | undefined {
    return this._etape;
  }

  public set etape(value: Etape | undefined) {
    this._etape = value;
  }
  */
  public get client(): Client | undefined {
    return this._client;
  }

  public set client(value: Client | undefined) {
    this._client = value;
  }

  public get participant(): Participant | undefined {
    return this._participant;
  }

  public set participant(value: Participant | undefined) {
    this._participant = value;
  }

  public get roadtrip(): Roadtrip | undefined {
    return this._roadtrip;
  }

  public set roadtrip(value: Roadtrip | undefined) {
    this._roadtrip = value;
  }
}
